var/spack/repos/builtin/packages/asio/package.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os.path
+
+
+class Asio(AutotoolsPackage):
+    """Asio is a cross-platform C++ library for network and low-level I/O programming."""
+
+    homepage = "http://think-async.com/Asio/"
+    url = "https://github.com/chriskohlhoff/asio/archive/1.18.2.tar.gz"
+    git = "https://github.com/chriskohlhoff/asio.git"
+    maintainers = ["msimberg", "???TODO???"]
+
+    version("1.18.2", sha256="42e6111a2185bdadb9feb8e3cf52e4c415c118b0")
+
+    depends_on("autoconf", type="build")
+    depends_on("automake", type="build")
+    depends_on("m4", type="build")
+    depends_on("libtool", type="build")
+
+    variant(
+        "cxxstd",
+        default="11",
+        values=("11", "14", "17"),
+        multi=False,
+        description="Use the specified C++ standard when building.",
+    )
+
+    variant(
+        "separate_compilation",
+        default=False,
+        description="Compile Asio sources separately (when disabled, Asio is header-only)",
+    )
+
+    variant(
+        "boost_coroutine",
+        default=False,
+        description="Enable support for Boost.Coroutine",
+    )
+    depends_on(
+        "boost +context +coroutine cxxstd=11",
+        when="+boost_coroutine cxxstd=11",
+    )
+    depends_on(
+        "boost +context +coroutine cxxstd=14",
+        when="+boost_coroutine cxxstd=14",
+    )
+    depends_on(
+        "boost +context +coroutine cxxstd=17",
+        when="+boost_coroutine cxxstd=17",
+    )
```

Comment:
Is there a more idiomatic way of doing this? `self.variants[cxxstd].value` does not seem to work here, probably because it doesn't have a value yet at this point?

Reply:
I don't think so :( but you can split the constraints on boost like this:

```
depends_on('boost +context +coroutine', when='+boost_coroutine')
depends_on('boost +regex', when='+boost_regex')

for std in ['11', '14', '17']:
  depends_on('boost cxxstd=' + std, when='cxxstd=' + std)
```

Reply:
That requires the new concretizer which I think is OK by now...

It is however not exactly equivalent, since `when='... ^boost'` may also match a transient dependency on boost instead of direct dependency

Reply:
Thanks @haampie! This seems to work even with the old concretizer, unless the default changed already on develop? At least I don't have `clingo` in my config.